src/components/chip/chip.tsx

```diff
@@ -147,41 +213,71 @@ export class Chip
   }
   //--------------------------------------------------------------------------
   //
-  //  Public Methods
+  //  Event Listeners
   //
   //--------------------------------------------------------------------------
 
-  /** When `closable` is `true`, sets focus on the component's "close" button (the first focusable item). */
-  @Method()
-  async setFocus(): Promise<void> {
-    await componentLoaded(this);
+  @Listen("keydown", { capture: true })
+  keyDownHandler(event: KeyboardEvent): void {
+    if (event.target === this.el) {
+      switch (event.key) {
+        case " ":
+        case "Enter":
+          this.itemSelectHandler();
+          event.preventDefault();
+          break;
+        case "ArrowRight":
+        case "ArrowLeft":
+        case "Home":
+        case "End":
+          this.calciteInternalChipKeyEvent.emit(event);
+          event.preventDefault();
+          break;
+      }
+    }
+  }
 
-    this.closeButton?.focus();
+  @Listen("calciteChipInternalSelectionChange", { target: "body" })
+  internalSelectionChangeListener(event: CustomEvent): void {
+    if (event.target !== this.parent) {
+      return;
+    }
+    this.determineActiveItem(event.detail);
```

Comment:
Not for this PR, but I think we should consider moving this logic to the group. This could apply to other components. I'll create an issue to look into this.

Reply:
Agreed - I can move that for this PR and clean up any unneeded internal events.